connection.php

```diff
@@ -0,0 +1,725 @@
+		// {
+		//     $office_street = test_input($input["office_street"]);
+		//     // checks for the presence of letter
+		//     if (!preg_match("/^[a-zA-Z]*$/", $office_street))
+		//     {
+		//         $_SESSION['error']['office_street_err'] = $error_list['office_street'];
+		//         $error++;
+		//     }
+		// }
+		// else
+		// {
+		//     $_SESSION['error']['office_street_err'] = $error_space['office_street'];
+		// }
+		if (isset($input["office_city"]))
+		{
+			$office_city = $this->test_input($input["office_city"]);
+			// checks for the presence of letter
+			if (!preg_match("/^[a-zA-Z]*$/", $office_city))
+			{
+				$_SESSION['error']['office_city_err'] = $error_list['office_city'];
+				$error++;
+			}
+		}
+		else
+		{
+			$_SESSION['error']['office_city_err'] = $error_space['office_city'];
+		}
+
+		if (empty($input["office_state"]))
+		{
+			$_SESSION['error']['office_state_err'] = $error_list['office_state'];
+			$error++;
+		}
+
+		if (isset($input["office_zip"]))
+		{
+			$office_zip = $this->test_input($input["office_zip"]);
+
+			// checks for the presence of letter
+
+			if (!preg_match("/^[0-9]*$/", $office_zip))
+			{
+				$_SESSION['error']['office_zip'] = $error_list['office_zip'];
+				$error++;
+			}
+		}
+		else
+		{
+			$_SESSION['error']['office_zip_err'] = $error_space['office_zip'];
+		}
+
+		// if (isset($input["office_mobile"]))
+		// {
+		//     $office_mobile = test_input($input["office_mobile"]);
+		//     // checks for the presence of letter
+		//     if (!preg_match("/^[0-9]*$/", $office_mobile))
+		//     {
+		//         $_SESSION['error']['office_mobile_err'] = $error_list['office_mobile'];
+		//         $error++;
+		//     }
+		// }
+		// else
+		// {
+		//     $_SESSION['error']['office_mobile_err'] = $error_space['office_mobile'];
+		// }
+		if (isset($input["office_landline"]))
+		{
+			$office_landline = $this->test_input($input["office_landline"]);
+			// checks for the presence of letter
+
+			if (!preg_match("/^[0-9]*$/", $office_landline))
+			{
+				$_SESSION['error']['office_landline_err'] = $error_list['office_landline'];
+				$error++;
+			}
+		}
+		else
+		{
+			$_SESSION['error']['office_landline_err'] = $error_space['office_landline'];
+		}
+
+		if (isset($input["office_fax"]))
+		{
+			$office_fax = $this->test_input($input["office_fax"]);
+			// checks for the presence of letter
+
+			if (!preg_match("/^[0-9]*$/", $office_fax))
+			{
+				$_SESSION['error']['office_fax'] = $error_list['office_fax'];
+				$error++;
+			}
+		}
+		else
+		{
+			$_SESSION['error']['office_fax_err'] = $error_space['office_fax'];
+		}
+
+		if (empty($input["communication"]))
+		{
+			$_SESSION['error']['communication_err'] = $error_list['communication'];
+			$error++;
+		}
+
+		return $error;
+
+
+	}
+
+	/*
+	 * @description: all string data are validated before form validation starts
+	 * @param: string
+	 * @return: string
+	*/
+
+	function test_input($data)
+    {
+        $data = trim($data);
```

Comment:
These code can be reduced to just below.

function test_input($data)
{
    return htmlspecialchars(stripslashes(trim($data)));
}